File: Source/Aladdin.CAPI.ANSI.PKCS11/MAC/HMAC_SHA1.cs
using System;
using Aladdin.PKCS11; 

namespace Aladdin.CAPI.ANSI.PKCS11.MAC
{
	///////////////////////////////////////////////////////////////////////////////
	// Алгоритм вычисления имитовставки HMAC SHA1
	///////////////////////////////////////////////////////////////////////////////
	public class HMAC_SHA1 : HMAC
	{
		// конструктор
        public HMAC_SHA1(CAPI.PKCS11.Applet applet) : this(applet, 20) {}

		// конструктор
		public HMAC_SHA1(CAPI.PKCS11.Applet applet, int macSize) 
            
            // сохранить переданные параметры
            : base(applet, API.CKM_SHA_1_HMAC, API.CKM_SHA_1, API.CKK_SHA_1_HMAC, macSize) {}
    }
}

File: Source/Aladdin.CAPI.ANSI.PKCS11.Java/src/aladdin/capi/ansi/pkcs11/mac/HMAC_SHA1.java
package aladdin.capi.ansi.pkcs11.mac;
import aladdin.capi.pkcs11.*; 
import aladdin.pkcs11.*;
import java.io.*; 

///////////////////////////////////////////////////////////////////////////////
// Алгоритм вычисления имитовставки HMAC SHA1
///////////////////////////////////////////////////////////////////////////////
public class HMAC_SHA1 extends HMAC
{
	// конструктор
	public HMAC_SHA1(Applet applet) throws IOException { this(applet, 20); }

	// конструктор
	public HMAC_SHA1(Applet applet, int macSize) throws IOException  
    { 
        // сохранить переданные параметры
        super(applet, API.CKM_SHA_1_HMAC, API.CKM_SHA_1, macSize); 
    } 
}
